api/v1beta2/foundationdbbackup_types.go

```diff
@@ -234,16 +249,27 @@ func (backup *FoundationDBBackup) Bucket() string {
 // BackupName gets the name of the backup in the destination.
 // This will fill in a default value if the backup name in the spec is empty.
 func (backup *FoundationDBBackup) BackupName() string {
-	if backup.Spec.BlobStoreConfiguration.BackupName == "" {
-		return backup.ObjectMeta.Name
+	if backup.Spec.BlobStoreConfiguration != nil && backup.Spec.BlobStoreConfiguration.BackupName != "" {
+		return backup.Spec.BlobStoreConfiguration.BackupName
+	}
+
+	if backup.Spec.FSConfiguration != nil && backup.Spec.FSConfiguration.BackupName != "" {
+		return backup.Spec.FSConfiguration.BackupName
 	}
 
-	return backup.Spec.BlobStoreConfiguration.BackupName
+	return backup.ObjectMeta.Name
 }
 
 // BackupURL gets the destination url of the backup.
 func (backup *FoundationDBBackup) BackupURL() string {
-	return backup.Spec.BlobStoreConfiguration.getURL(backup.BackupName(), backup.Bucket())
+	if backup.Spec.BlobStoreConfiguration != nil {
+		return backup.Spec.BlobStoreConfiguration.getURL(backup.BackupName(), backup.Bucket())
+	}
+	// mpatou: should we use file:// ?
+	if backup.Spec.FSConfiguration != nil {
+		return fmt.Sprintf("%s/%s", backup.Spec.FSConfiguration.URL, backup.BackupName())
```

Comment:
if this is a filesystem path, you should use
```suggestion
		return path.Join(backup.Spec.FSConfiguration.URL, backup.BackupName())
```